pkg/config/filter.go

```diff
@@ -23,5 +23,5 @@ type FilterConfig struct {
 	Rules []string `toml:"rules" json:"rules"`
 	*filter.MySQLReplicationRules
 	IgnoreTxnStartTs []uint64           `toml:"ignore-txn-start-ts" json:"ignore-txn-start-ts"`
-	DDLAllowlist     []model.ActionType `toml:"ddl-allow-list" json:"ddl-allow-list"`
+	DDLAllowlist     []model.ActionType `toml:"ddl-allow-list" json:"ddl-allow-list,omitempty"`
```

Comment:
I think we should omit the field always not just when `ddl-allow-list` empty.

Reply:
but if the we set `json:"-"` for `ddl-allow-list`, the `DDLAllowlist` will be not persistent in ETCD, so this will make `ddl-allow-list` ineffective and break the compatibility

Reply:
Can we implement the fmt.Stringer interface to hide this field when output logs.

Reply:
If a user already sets the `ddl-allow-list`, it's no need to hide `ddl-allow-list` for him, because he has already known it. On the other hand, we need to observe `ddl-allow-list` in the log for debugging.
If a user doesn't know `ddl-allow-list`, the config entry will be default value (nil), so he can't know ``ddl-allow-list``